### PR TITLE
Remove serialization of SourceTexts

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/72705

All non-test code already is properly only dealing with a SerializableSoruceText.  Fixed up the tests to do the same, and now we can remove these extra cases from teh product code.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal.

File: src/Workspaces/Core/Portable/Remote/WellKnownSynchronizationKind.cs [hunk 1]
```diff
@@ -28,7 +28,6 @@ internal enum WellKnownSynchronizationKind
     ProjectReference,
     MetadataReference,
     AnalyzerReference,
-    SourceText,
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Remote/WellKnownSynchronizationKind.cs:31]: removed this kind entirely.  the only thing we sujpport serializing are SerializableSourceText

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**